### PR TITLE
[Kotlin Migration] ActivityExportingDelegate

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt [hunk 1]
```diff
@@ -13,206 +13,178 @@
...
-            saveExportFile(path);
+            showSimpleSnackbar(activity, R.string.export_send_no_handlers, false)
+            saveExportFile(path)
         }
     }
 
-
-    @Override
-    public void saveExportFile(String path) {
+    override fun saveExportFile(exportPath: String?) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt:125]: cleanup: non-null

File: AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt [hunk 2]
```diff
@@ -13,206 +13,178 @@
...
+        val isSuccessful = exportToProvider(result.data, true)
+        @StringRes val message = if (isSuccessful) R.string.export_save_apkg_successful else R.string.export_save_apkg_unsuccessful
+        showSimpleSnackbar(activity, activity.getString(message), isSuccessful)
     }
 
-    private boolean exportToProvider(Intent intent, boolean deleteAfterExport) {
-        if ((intent == null) || (intent.getData() == null)) {
-            Timber.e("exportToProvider() provided with insufficient intent data %s", intent);
-            return false;
+    private fun exportToProvider(intent: Intent?, deleteAfterExport: Boolean): Boolean {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt:152]: I'd be tempted to make this non-null

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Few cleanups. Seems good though

**Review by david-allison-1 — APPROVED** [review]
LGTM, thanks

**Review by mikehardy — APPROVED** [review]
:sweat_smile: lots of tech debt was embodied in those nullables, better that it's obvious at least, thanks!

**Pull request merged**